hack/update-mocks.sh

```diff
@@ -84,14 +100,14 @@ for IFILE in $(find_files | xargs grep --files-with-matches -e '//go:generate mo
 done
 
 
-# get the changed mock files
-files=$(git diff --name-only)
+# get the changed or new mock files
+files=$(git status --porcelain --untracked-files | awk '{print $NF}')
 for file in $files; do
-  if [ "$file" == "hack/update-mocks.sh" ]; then
-    continue
-  fi
+  # only process files that appear to be mocks
+  test -f "$file" || continue
+  grep -q "^// Code generated by MockGen. DO NOT EDIT.$" "$file" || continue
```

Comment:
would limiting this to .go files and fuzzy matching the comment (e.g. "^//.*generated by MockGen") be less fragile?

Reply:
Note: this is only evaluating modifed or new (not tracked) files, which SHOULD only be the newly generate mocks.  Not sure extra fuzziness helps - that is the string being used and if it changes I think we WANT to know..  At least one file looks like it was generated by the tool and committed without a regen possible.

Reply:
ok... will anything notice if the mockgen comment changes and this script doesn't change?

Reply:
Assuming we bumped the depency to a version thast used a different string, CI would fail verify-mocks.sh, so we'd at least have to go and regenerate all the mockfiles, presumably with this script.  Since the comment would have changed, this script will not find them and will therefore not add boilerplate.  So even if someone bypassed all that and then commited them, it would fail boilerplate.

So ... "yes, I think so" :)